### PR TITLE
Additional mutation types

### PR DESCRIPTION
**Pull request by Noctifer-de-Mortem**
#### Summary

```SUMMARY: Content "Implements additional mutation types."```

#### Purpose of change

Partially expands the JSON usage of mutation types as initially founded in https://github.com/CleverRaven/Cataclysm-DDA/pull/25201, applying it to more mutations to expand mod compatibility.

#### Describe the solution

This applies mutation types WINGS, TEETH, MUZZLES, LEGS, TAILS, HORNS, EARS, and WHISKERS while attempting to retain intended status quo of cancellations.

WINGS applies to mutations that affect the back in general (vines and similar cancel), name chosen to reduce chance of merge conflicts with https://github.com/CleverRaven/Cataclysm-DDA/pull/25582.

TEETH and MUZZLES allow for untangling the often-confusing cancellations between various mouth-related mutations. A notable change is removing the situation where beaks cancel mouth tentacles, but not vice-versa. It seems that mimicking a squid-like appearance is intended due to beak being available to cephalopod mutants, so this one-way cancellation has been removed. No type is currently set up for mouth tentacles as they could possible be compatible with other mouth mutations (see squids, star-faced moles, etc), but if needed a category for them (not clashing with beak) can be devised.

LEGS covers incompatible feet and leg mutations as a whole. TAILS, EARS, WHISKERS, and HORNS are likewise basic expansions of the system, though HORNS include antennae too.

#### Describe alternatives you've considered

Content addition so no alternatives in play.

#### Additional context

Context has been provided above.

### CODE REVIEW

**Comment by Noctifer-de-Mortem** [comment]
> 23:49:50 Invalid Summary line, please provide a summary similar to the following:
> 23:49:50 SUMMARY: Features "Added ability to fly."

Summary looks fine from what I see. Anything I missed.

**Comment by alanbrady** [comment]
Jenkins rebuild

**Comment by alanbrady** [comment]
6k LOC diff what?  It looks like you styled the json I guess, but it's all one giant commit.  I have no idea what's going on here.

**Comment by AMurkin** [comment]
> Anything I missed.
Should be 0 or 3 apostrophes around summary line.

**Comment by Noctifer-de-Mortem** [comment]
> 
> 
> > Anything I missed.
> 
> Should be 0 or 3 apostrophes around summary line.

Right right, thank you. Still new to new PR definitions so appreciate the help.

**Comment by ZhilkinSerg** [comment]
> 6k LOC diff what? It looks like you styled the json I guess, but it's all one giant commit. I have no idea what's going on here.

Yeah, that was mostly linted json - https://github.com/CleverRaven/Cataclysm-DDA/pull/25670/files?w=1

**Comment by ZhilkinSerg** [comment]
Jenkins rebuild.

**Comment by alanbrady** [comment]
> Yeah, that was mostly linted json
Which is probably fine, but I think it should be in at the very least a separate commit if not an entirely separate PR if this file needs to be styled.  It's impossible to tell what's actually been changed with the way this has been done.

**Comment by Noctifer-de-Mortem** [comment]
Hmm, thing is, I did not run anything through the linter, just added the new changes to the file and it seems it wants to replace the entire thing... Any idea why it might choose to do that?

**Comment by AMurkin** [comment]
Endlines changed: `0a -> 0a0d`.

image

**Comment by Noctifer-de-Mortem** [comment]
@AMurkin 
How do I go about fixing that? I tried replacing spaces but that didn't do anything.

**Comment by AMurkin** [comment]
Github page about line endings: https://help.github.com/articles/dealing-with-line-endings/
Or use advanced text editor which have command to set line endings type.

**Comment by Kelenius** [comment]
Okay, this is probably a nitpick, but it bothers me, and it would be hard to change later. I think the IDs for the types should be singular, like `SKIN` and `SCENT`. So `MUZZLES` should be `MUZZLE` and `TAILS` should be `TAIL`. "Mutation type is MUZZLE" makes more sense to me than "mutation type is MUZZLES", and it gives consistency.

**Comment by Noctifer-de-Mortem** [comment]
Wait, I messed up... changed the mutation type pluralization but not in the file...

**Comment by AMurkin** [comment]
Tests are failing.
```
-------------------------------------------------------------------------------
Having all pre-threshold mutations gives a sensible threshold breach chance
     Given: The player has all pre-threshold mutations for LUPINE
      Then: Threshold breach chance is at least 0.2
-------------------------------------------------------------------------------
mutation_test.cpp:111
...............................................................................
mutation_test.cpp:113: FAILED:
  CHECK( breach_chance >= BREACH_CHANCE_MIN )
with expansion:
  0.19798f >= 0.2f
with message:
  MUTATIONS: CLAWS HEAVYSLEEPER2 MUZZLE ANIMALDISCORD SMELLY TAIL_FLUFFY FANGS
  PALE GOODCARDIO PAWS GOURMAND NIGHTVISION2 SHOUT3 LUPINE_FUR PADDED_FEET
  TAIL_LONG STR_UP_3 SNARL LUPINE_EARS DEFORMED 
```

**Comment by AMurkin** [comment]
I think so. Tests were introduced in #21911.

File: data/json/mutations.json [hunk 1]
```diff
@@ -2678,9 +2679,8 @@
         "visibility" : 6,
         "ugliness" : 2,
         "description" : "You have a long, graceful tail, like that of a big cat.  It improves your balance, making your ability to dodge higher, but prevents wearing non-fabric pants.",
+        "types" : ["TAIL"],
         "prereqs" : ["TAIL_STUB"],
-        "cancels" : ["TAIL_FIN", "TAIL_THICK", "TAIL_RAPTOR"],
-        "changes_to" : ["TAIL_FLUFFY", "TAIL_STING", "TAIL_CLUB", "TAIL_CATTLE", "TAIL_RAT"],
```
Comment (Kelenius) [inline data/json/mutations.json]: I suspect that removal of this line is what causes the issue.

**Comment by kevingranade** [comment]
That test is trying to verify that a player has a decent change of breaching the threshold given proper preparation, dropping below that chance means the post-threshold mutations will be difficult to achieve, and dropping far below it might mean it's impossible to reach.

**Pull request merged**